install/install.ps1

```diff
@@ -8,87 +8,196 @@ $script:ErrorActionPreference='Stop'
 # We don't currently have good cross-platform options for determining the latest release version, so we require
 # knowledge of the specific target version, which the user should set as an environment variable.
 $version = $Env:AZUREAUTH_VERSION
+#$version = "0.3.1"
 if ([string]::IsNullOrEmpty($version)) {
     Write-Error 'No $AZUREAUTH_VERSION specified, unable to download a release'
 }
 
-$repo = if ([string]::IsNullOrEmpty($Env:AZUREAUTH_REPO)) { 'AzureAD/microsoft-authentication-cli' } else { $Env:AZUREAUTH_REPO }
-$releaseName = "azureauth-${version}-win10-x64"
-$releaseFile = "${releaseName}.zip"
-$releaseUrl = "https://github.com/${repo}/releases/download/${version}/$releaseFile"
 
-$azureauthDirectory = if ([string]::IsNullOrEmpty($Env:AZUREAUTH_INSTALL_DIRECTORY)) {
-    ([System.IO.Path]::Combine($Env:LOCALAPPDATA, "Programs", "AzureAuth"))
-} else {
-    $Env:AZUREAUTH_INSTALL_DIRECTORY
-}
-$extractedDirectory = ([System.IO.Path]::Combine($azureauthDirectory, $releaseName))
-$latestDirectory = ([System.IO.Path]::Combine($azureauthDirectory, "latest"))
-$zipFile = ([System.IO.Path]::Combine($azureauthDirectory, $releaseFile))
-
-Write-Verbose "Creating ${azureauthDirectory}"
-$null = New-Item -ItemType Directory -Force -Path $azureauthDirectory
-
-Write-Verbose "Downloading ${releaseUrl} to ${zipFile}"
-$client = New-Object System.Net.WebClient
-$client.DownloadFile($releaseUrl, $zipFile)
-
-# A running instance of azureauth can cause installation to fail, so we try to kill any running instances first.
-# We suppress taskkill output here because this is a best effort attempt and we don't want the user to see its output.
-# Here, Get-Process is used to first determine whether there is an existing azureauth process. If there is, kill the existing process first.
-$ProcessCheck = Get-Process -Name azureauth -ErrorAction SilentlyContinue -ErrorVariable ProcessError
-if ($ProcessCheck -ne $null)
-{
-    Write-Verbose "Stopping any currently running azureauth instances"
-    taskkill /f /im azureauth.exe 2>&1 | Out-Null
-
-    # After killing the process it is still possible for there there to be locks on the files it was using (including
-    # its own DLLs). The OS may take an indeterminate amount of time to clean those up, but so far we've observed 1
-    # second to be enough.
-    Start-Sleep -Seconds 1
-}
 
-if (Test-Path -Path $extractedDirectory) {
-    Write-Verbose "Removing pre-existing extracted directory at ${extractedDirectory}"
-    Remove-Item -Force -Recurse $extractedDirectory
-}
+function V1-Install-With-Latest{

```

Comment:
The `V1` vs. `V2` distinction is a little muddy. Let's just leave this as `Install-With-Latest` and make the opposite be `Install-Without-Latest`. I would also be happy to see this be named something like `Install-Pre-0-4-0` and `Install-Post-0-4-0` or similar if PowerShell allows a function name like that. Thoughts here, @kyle-rader?